sympy/utilities/pytest.py

```diff
@@ -90,6 +91,43 @@ def raises(expectedException, code=None):
             raise TypeError(
                 'raises() expects a callable for the 2nd argument.')
 
+    def assert_raise_message(expectedException, err_msg, code=None):
+        """
+        ``assert_raise_message`` is an enhanced version of ``raises``. It tests
+        that error message raised by ``code`` is also same as the expected error
+        message.
+
+        ``assert_raise_message()`` does nothing if the callable raises the expected
+        ``err_msg``, otherwise it raises an AssertionError.
+
+        Examples
+        ========
+
+        >>> from sympy.utilities.pytest import assert_raise_message
+
+        >>> assert_raise_message(ZeroDivisionError, 'zero', lambda: 1/0)
+        >>> assert_raise_message(ZeroDivisionError, 'one', lambda: 1/0)
+        Traceback (most recent call last):
+        ...
+        AssertionError: DID NOT RAISE
+
+        """
+        if code is None:
+            return RaisesContext(expectedException)
+        elif callable(code):
+            try:
+                code()
+            except expectedException as message:
+                message = message.args[0]
+                found = re.search(err_msg, message, re.M|re.I)
```

Comment:
It would be nice for `raises` or `assert_raise_message` to accept a compiled `pattern`:
```python
import re
ci_pattern = re.compile(r"Case doesn't matter", re.IGNORECASE)
raises(Exception, code=code, err_msg=ci_pattern)
```
and a function that receives exception as an argument and returns a `bool`:
```python
raises(Exception, code=code, check=lambda x: isinstance(x, TypeError))
```

Reply:
@ylemkimon @gxyd I didn't understand what is told here. Should I make a new routine `rasiesExceprtion` ? or should I only use the compiled pattern to use for search operation.

Reply:
I was thinking adding it to `raises`, but it would break `pytest` compatibility, so you can add them to `assert_raise_message`.